src/js/fable-core/Set.fs

```diff
@@ -603,80 +603,81 @@ let iterate f (s : Set<'T>)  = s.Iterate(f)
+let fold<'T,'State  when 'T : comparison> f (z: 'State) (s: Set<'T>) = SetTree.fold f z s.Tree
 
-let foldBack<'T,'State when 'T : comparison> f (s : Set<'T>) (z:'State) = SetTree.foldBack f s.Tree z
+let foldBack<'T,'State when 'T : comparison> f (s: Set<'T>) (z:'State) = SetTree.foldBack f s.Tree z
 
-let map f (s : Set<'T>) ([<Inject>] comparer: IComparer<'U>): Set<'U> =
+let map f (s: Set<'T>) ([<Inject>] comparer: IComparer<'U>): Set<'U> =
     new Set<_>(comparer, SetTree.fold (fun acc k -> SetTree.add comparer (f k) acc) (SetTree<_>.SetEmpty) s.Tree)
 
-let count (s : Set<'T>) = s.Count
+let count (s: Set<'T>) = s.Count
 
-let minimumElement (s : Set<'T>) = s.MinimumElement
+let minimumElement (s: Set<'T>) = s.MinimumElement
 
-let maximumElement (s : Set<'T>) = s.MaximumElement
+let maximumElement (s: Set<'T>) = s.MaximumElement
 
-let ofList (li : 'T list) ([<Inject>] comparer: IComparer<'T>) : Set<'T> =
+let ofList (li: 'T list) ([<Inject>] comparer: IComparer<'T>) : Set<'T> =
     new Set<_>(comparer, SetTree.ofSeq comparer li)
 
-let ofArray (arr : 'T array) ([<Inject>] comparer: IComparer<'T>) : Set<'T> =
+let ofArray (arr: 'T array) ([<Inject>] comparer: IComparer<'T>) : Set<'T> =
     new Set<_>(comparer, SetTree.ofArray comparer arr)
 
-let toList (s : Set<'T>) = SetTree.toList s.Tree
+let toList (s: Set<'T>) = SetTree.toList s.Tree
 
-let toArray (s : Set<'T>) ([<Inject>] cons: Array.IArrayCons<'T>) =
+let toArray (s: Set<'T>) ([<Inject>] cons: Array.IArrayCons<'T>) =
     let n = (count s)
     let res = cons.Create n
     SetTree.copyToArray s.Tree res 0
     res
 
-let toSeq (s : Set<'T>) =
+let toSeq (s: Set<'T>) =
     SetTree.toSeq s.Tree
 
-let ofSeq (elements : seq<'T>) ([<Inject>] comparer: IComparer<'T>) =
+let ofSeq (elements: seq<'T>) ([<Inject>] comparer: IComparer<'T>) =
     new Set<_>(comparer, SetTree.ofSeq comparer elements)
 
 let difference (x: Set<'T>) (y: Set<'T>) = x - y
 
-let isSubset (x:Set<'T>) (y: Set<'T>) = x.IsSubsetOf(y)
+let isSubset (x: Set<'T>) (y: Set<'T>) = x.IsSubsetOf(y)
 
-let isSuperset (x:Set<'T>) (y: Set<'T>) = x.IsSupersetOf(y)
+let isSuperset (x: Set<'T>) (y: Set<'T>) = x.IsSupersetOf(y)
 
-let isProperSubset (x:Set<'T>) (y: Set<'T>) = x.IsProperSubsetOf(y)
+let isProperSubset (x: Set<'T>) (y: Set<'T>) = x.IsProperSubsetOf(y)
 
-let isProperSuperset (x:Set<'T>) (y: Set<'T>) = x.IsProperSupersetOf(y)
+let isProperSuperset (x: Set<'T>) (y: Set<'T>) = x.IsProperSupersetOf(y)
 
-let minElement (s : Set<'T>) = s.MinimumElement
+let minElement (s: Set<'T>) = s.MinimumElement
 
-let maxElement (s : Set<'T>) = s.MaximumElement
+let maxElement (s: Set<'T>) = s.MaximumElement
 
-// let create (l : seq<'T>) = Set<_>.Create(l)
+// let create (l: seq<'T>) = Set<_>.Create(l)
 
 type IMutableSet<'T> =
     inherit IEnumerable<'T>
     abstract size: int
-    abstract add: 'T -> IMutableSet<'T>
+    abstract add: 'T -> bool
```

Comment:
@ncave Do you need this? We'll probably have to revert this change because the `IMutableSet` interface is intended to match [JS Set](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set/add) (I should have left a comment about this, JS Set is used to emulate .NET HashSet, however when the key has custom comparison we call `createMutable` to emulate the JS Set interface with an F# Set). Returning bool is actually [done by `Util/addToSet`](https://github.com/fable-compiler/Fable/blob/43768e73f3cd9957d312e3788d3fcece98f6df04/src/dotnet/Fable.Compiler/Transforms/Replacements.fs#L1739-L1740).